### PR TITLE
sxDai Tests

### PR DESCRIPTION
**Pull request by jrhea**
Adding tests for sxDai deployment on Gnosis

### CODE REVIEW

File: test/instances/erc4626/sxDai.t.sol [hunk 1]
```diff
@@ -0,0 +1,755 @@
...
+contract sxDaiHyperdriveTest is InstanceTest {
+    using FixedPointMath for uint256;
+    using HyperdriveUtils for uint256;
+    using HyperdriveUtils for IHyperdrive;
+    using Lib for *;
+    using stdStorage for StdStorage;
+
+    // The xDai contract.
+    IERC20 internal constant XDAI =
+        IERC20(0xe91D153E0b41518A2Ce8Dd3D7944Fa863463a97d);
```
Comment (jalextowle) [inline test/instances/erc4626/sxDai.t.sol:37]: What do you think about calling this `wxDAI`?

Comment (jrhea) [reply test/instances/erc4626/sxDai.t.sol:37]: ya ill change it

File: test/instances/erc4626/sxDai.t.sol [hunk 2]
```diff
@@ -0,0 +1,755 @@
...
+    // The wxDai contract.
+    IERC20 internal constant WXDAI =
+        IERC20(0xe91D153E0b41518A2Ce8Dd3D7944Fa863463a97d);
+
+    // The sxDai contract.
+    IERC4626 internal constant SXDAI =
+        IERC4626(0xaf204776c7245bF4147c2612BF6e5972Ee483701);
+
+    // Whale accounts.
+    address internal XDAI_TOKEN_WHALE =
```
Comment (jalextowle) [inline test/instances/erc4626/sxDai.t.sol:44]: Nit: Let's call this the `WXDAI_TOKEN_WHALE`

Comment (jrhea) [reply test/instances/erc4626/sxDai.t.sol:44]: doh

File: test/instances/erc4626/sxDai.t.sol [hunk 3]
```diff
@@ -0,0 +1,755 @@
...
+            false
+        );
+        uint256 baseProceeds = convertToBase(shareProceeds);
+
+        // Bob should receive almost exactly the interest that accrued on the
+        // bonds that were shorted.
+        assertApproxEqAbs(
+            baseProceeds,
+            _shortAmount.mulDown(_variableRate),
+            1e9
```
Comment (jalextowle) [inline test/instances/erc4626/sxDai.t.sol:721]: I was able to reduce this a lot with 10k fuzz runs:

```suggestion
            1e3
```

File: test/instances/erc4626/sxDai.t.sol [hunk 4]
```diff
@@ -0,0 +1,755 @@
...
+            maturityTime,
+            _shortAmount,
+            false
+        );
+        uint256 baseProceeds = convertToBase(shareProceeds);
+
+        // Bob should receive approximately as much base as he paid since no
+        // time as passed and the fees are zero.
+        assertLt(baseProceeds, basePaid + 200);
+        assertApproxEqAbs(baseProceeds, basePaid, 1e9);
```
Comment (jalextowle) [inline test/instances/erc4626/sxDai.t.sol:664]: I was able to reduce this a lot:

```suggestion
        assertApproxEqAbs(baseProceeds, basePaid, 1e3);
```

File: test/instances/erc4626/sxDai.t.sol [hunk 5]
```diff
@@ -0,0 +1,755 @@
...
+            0,
+            IHyperdrive.Options({
+                destination: bob,
+                asBase: true,
+                extraData: new bytes(0)
+            })
+        );
+
+        // Ensure that Bob receives a refund when he opens a short with "asBase"
+        // set to false and sends ether to the contract.
```
Comment (jalextowle) [inline test/instances/erc4626/sxDai.t.sol:614]:
```suggestion
        // Ensure that sending ETH to `openShort` fails with `asBase` as false.

```

File: test/instances/erc4626/sxDai.t.sol [hunk 6]
```diff
@@ -0,0 +1,755 @@
...
+            1e9
+        );
+    }
+
+    /// Long ///
+
+    function test_open_long_nonpayable() external {
+        vm.startPrank(bob);
+
+        // Ensure that sending ETH to `openLong` fails.
```
Comment (jalextowle) [inline test/instances/erc4626/sxDai.t.sol:481]:
```suggestion
        // Ensure that sending ETH to `openLong` fails with `asBase` as true.
```

File: test/instances/erc4626/sxDai.t.sol [hunk 7]
```diff
@@ -0,0 +1,755 @@
...
+            0,
+            0,
+            IHyperdrive.Options({
+                destination: bob,
+                asBase: true,
+                extraData: new bytes(0)
+            })
+        );
+
+        // Ensure that sending ETH to `openShort` fails.
```
Comment (jalextowle) [inline test/instances/erc4626/sxDai.t.sol:494]:
```suggestion
        // Ensure that sending ETH to `openLong` fails with `asBase` as false.
```

File: test/instances/erc4626/sxDai.t.sol [hunk 8]
```diff
@@ -0,0 +1,755 @@
...
+        );
+        baseProceeds = convertToBase(shareProceeds);
+        assertEq(withdrawalSharesRedeemed, withdrawalShares);
+
+        // Bob should receive base approximately equal in value to his present
+        // value.
+        assertApproxEqAbs(
+            baseProceeds,
+            withdrawalShares.mulDown(lpSharePrice),
+            1e9
```
Comment (jalextowle) [inline test/instances/erc4626/sxDai.t.sol:472]: I was able to reduce this:

```suggestion
            1e5
```

File: test/instances/erc4626/sxDai.t.sol [hunk 9]
```diff
@@ -0,0 +1,755 @@
...
+            bob,
+            lpShares,
+            false
+        );
+        uint256 baseProceeds = convertToBase(shareProceeds);
+        assertEq(withdrawalShares, 0);
+
+        // Bob should receive approximately as much base as he contributed since
+        // no time as passed and the fees are zero.
+        assertApproxEqAbs(baseProceeds, _contribution, 1e10);
```
Comment (jalextowle) [inline test/instances/erc4626/sxDai.t.sol:406]: I was able to reduce this:

```suggestion
        assertApproxEqAbs(baseProceeds, _contribution, 1e5);
```

**Review by jalextowle — APPROVED** [review]
LGTM once the last few comments are addressed

**Pull request merged**